innkeeper/clientauth.go

```diff
@@ -0,0 +1,32 @@
+package innkeeper
+
+import (
+	"github.com/zalando/skipper/oauth"
+)
+
+// An Authentication object provides authentication to Innkeeper.
+type Authentication interface {
```

Comment:
well, that's ok. About srp and dip, to me it feels that the current one is more both srp and dip, since currently the innkeeper client doesn't directly depend on the oauth package (only in the documentation :))